api/history.go

```diff
@@ -0,0 +1,66 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	apitypes "github.com/puppetlabs/wash/api/types"
+	"github.com/puppetlabs/wash/journal"
+	log "github.com/sirupsen/logrus"
+)
+
+var historyHandler handler = func(w http.ResponseWriter, r *http.Request) *errorResponse {
+	history := journal.History()
+
+	if index := mux.Vars(r)["index"]; index == "" {
```

Comment:
Hmm, may be nice to split this into `historyHandler`/`historyEntryHandler`, respectively.